runner/android_test_orchestrator/javatests/androidx/test/orchestrator/AndroidTestOrchestratorTest.java

```diff
@@ -44,4 +44,32 @@ public void testSingleMethodTest_blankInput() {
     assertThat(AndroidTestOrchestrator.isSingleMethodTest(null), is(false));
     assertThat(AndroidTestOrchestrator.isSingleMethodTest(""), is(false));
   }
+
+  @Test
+  public void testMakeValidFilename_notTooLong() {
+    final int maxLength = 20;
+
+     assertThat(
+        // len("hello_world") = 11, with ".txt" appended it's still way below maxLength
+        AndroidTestOrchestrator.makeValidFilename("hello_world", maxLength),
+        is("hello_world.txt"));
+
+      assertThat(
+        // len("hello_world_abcd") = 16, with ".txt" appended it's equal to maxLength
+        AndroidTestOrchestrator.makeValidFilename("hello_world_abcd", maxLength),
+        is("hello_world_abcd.txt"));
+  }
+
+  @Test
+  public void testMakeValidFilename_tooLong() {
+      final int maxLength = 20;
+
+      final String testHashCode = String.format("%08x", "hello_world_abcde".hashCode());
+      assertThat(testHashCode, is("889cc989"));
```

Comment:
No need for this assert. We're not testing String.format or hashCode() here.

Reply:
I added it here to make it obvious to the reader what's the hash code of `hello_world_abcde`, so the later assert is more obvious to them.

Reply:
A code comment for that above testHashCode is plenty.

Reply:
Sure, will do!

Reply:
done